LibVLCSharp/Shared/Picture.cs

```diff
@@ -0,0 +1,127 @@
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_release")]
+            internal static extern void LibVLCPictureRelease(IntPtr picture);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_save")]
+            internal static extern int LibVLCPictureSave(IntPtr picture, IntPtr path);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_get_buffer")]
+            internal static extern ReadOnlyMemory<byte> LibVLCPictureGetBuffer(IntPtr picture, UIntPtr size);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_type")]
+            internal static extern PictureType LibVLCPictureType(IntPtr picture);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_get_stride")]
+            internal static extern uint LibVLCPictureGetStride(IntPtr picture);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_get_width")]
+            internal static extern uint LibVLCPictureGetWidth(IntPtr picture);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_get_height")]
+            internal static extern uint LibVLCPictureGetHeight(IntPtr picture);
+
+            [DllImport(Constants.LibraryName, CallingConvention = CallingConvention.Cdecl,
+                EntryPoint = "libvlc_picture_get_time")]
+            internal static extern long LibVLCPictureGetTime(IntPtr picture);
+        }
+
+        /// <summary>
+        /// Saves this picture to a file. The image format is the same as the one
+        /// returned by Type
+        /// </summary>
+        /// <param name="path">The path to the generated file</param>
+        /// <returns>true in case of success, false otherwise</returns>
+        public bool Save(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+                return false;
+
+            return Native.LibVLCPictureSave(NativeReference, path.ToUtf8()) == 0;
+        }
+
+        // TODO: test this, we probably don't want to expose UIntPtr as a param
+        /// <summary>
+        /// The image internal buffer, including potential padding.
+        /// The Picture owns the returned buffer, which must not be modified nor freed
+        /// </summary>
+        /// <param name="size">A pointer to a size_t that will hold the size of the buffer</param>
+        /// <returns>A pointer to the internal buffer.</returns>
+        public ReadOnlyMemory<byte> Buffer(UIntPtr size) => Native.LibVLCPictureGetBuffer(NativeReference, size);
```

Comment:
Not sure we can do that directly, the ReadOnlyMemory has no way to know the size.
I found this : https://github.com/dotnet/runtime/issues/29530 and that https://github.com/dotnet/corefx/blob/0d3caeb/src/Common/src/System/Memory/PointerMemoryManager.cs , but it needs to use ` unsafe` code. Too bad...

I found `SafeBuffer`, but the doc says it's deprecated, but didn't find why anywhere : https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safebuffer?view=netcore-3.1

Reply:
Anyway, the size pointer is probably a bad idea, it should be a property of what gets returned.